### PR TITLE
Providing initial test

### LINKED ISSUE #7: Provide first basic test checking that the Kaoto UI is displayed
- setup structure to play test
- basic scenario:
   - open a my.kaoto.yaml file
   - check that the UI is correctly loaded

### PR DESCRIPTION
**Pull request by apupier**
fixes #7

requires https://github.com/KaotoIO/vscode-kaoto/pull/32

### CODE REVIEW

**Comment by apupier** [comment]
current failure is due to the fact that kaoto-ui is not published and vscode-extension-tester is running a `yarn list --prod --json` which is not taking care of potential `yarn link`ed dependencies (at least if there isn't an initial published version)
asked if we can find a workaround https://coreos.slack.com/archives/C047NGXPRG8/p1666342896003189

**Comment by apupier** [comment]
the test itself is very very minimal. It is testing only that an element
from the top bar and one from the Canvas are loading. I think it is
enough to catch main regressions and validate a first iteration of the
test structure. of course, more advanced use cases will need to be
covered.

Technical:
- use xvfb github action. It is required for an unknown reason. Several
other VS Code test extension are not using it. Note that it is required
for Linux only.
- there is a long wait condition, it is the time needed for docker image
to be downloaded.
- remove Windows and MacOS because cannot play test on them due to
Docker issue:
    - On MacOS, Docker not available at all
    - On Windows: docker: image operating system "linux" cannot be used
on this platform.

File: it-tests/BasicFlow.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		await VSBrowser.instance.openResources(workspaceFolder);
+	});
+
+	after(async function() {
+		const editorView = new EditorView();
+		await editorView.closeAllEditors();
+	});
+
+	it('Open "empty.kaoto.yaml" file and check Kaoto UI is loading', async function () {
+		this.timeout(60000);
```
Comment (kahboom) [inline it-tests/BasicFlow.test.ts:24]: Does the docker image need to be downloaded for each assertion? Or why are we having so many different timeouts?

File: it-tests/BasicFlow.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		await checkPartOfTopBarLoaded(driver);
+		await checkCanvasLoaded(driver);
+		await kaotoWebview.switchBack();
+		assert.isFalse(await kaotoEditor.isDirty(), 'The Kaoto editor should not be dirty after everything has loaded.');
+	});
+
+});
+
+async function checkCanvasLoaded(driver: WebDriver) {
+	await driver.wait(until.elementLocated(By.xpath("//div[text()='ADD A STEP']")));
```
Comment (kahboom) [inline it-tests/BasicFlow.test.ts:46]: can we not use normal HTML selectors to check for the existence of something in the DOM, or is there a reason why we are using the driver + xpath? sorry I'm not too familiar with this approach so I may just need some enlightenment :)

File: it-tests/BasicFlow.test.ts (hunk 1 above)
Comment (apupier) [reply it-tests/BasicFlow.test.ts:24]: Need to be downloaded only one time.

This timeout is globally for the test execution.
We need to specify timeout for some specific driver.wait as default timeout for some is at 5s (if i remember well). The sum of all timeout in driver.wait should be lower than the global one.

File: it-tests/BasicFlow.test.ts (hunk 2 above)
Comment (apupier) [reply it-tests/BasicFlow.test.ts:46]: I used one way that is working. From my  far away experience several years ago), it was good habit to use either a `By.id` or a `By.xpath`. I'm open to use something else if it is better.

What is an HTML Selector?
The `By.name` https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_By.html ?

Comment (kahboom) [reply it-tests/BasicFlow.test.ts:46]: An HTML selector, at least in this case, would be the other functions available, where you would select by the native HTML elements (e.g. by `id`, by `class`). I think if you are more comfortable with `By.xpath` then that should be fine too, it's simply that I am not used to it. :)

**Review by kahboom — APPROVED** [review]

**Pull request merged**